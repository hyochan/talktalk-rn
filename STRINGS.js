import LocalizedStrings from 'react-native-localization';

export const strings = new LocalizedStrings({
  en: {
    LOADING: 'LOADING...',
    HELLO: 'Hello',
    LOGIN: 'Login',
    EMAIL: 'Email',
    PASSWORD: 'Password',
    SIGN_UP: 'Sign Up',
    LOGIN: 'Login',
    LOGOUT: 'Logout',
    FORGOT_PW: 'Forgot password?',
    FIND_PW: 'Find Password',
    NAME: 'name',
    STATUS_MSG: 'Status message',
    REGISTER: 'Register',
    SEND_LINK: 'Send Link',
    FRIEND: 'Friend',
    MESSAGE: 'Message',
    MY_PROFILE: 'My profile',
    UPDATE: 'Update',
    NO_CONTENT: 'No content',
    ADD_FRIEND: 'ADD',
    DELETE_FRIEND: 'DELETE',
    SEARCH_USER: 'Search User',
    CHAT: 'CHAT',
    SEND: 'Send',
    WRITE_MESSAGE: 'Write message...',
    ERROR: 'Error',
    FRIEND_ADDED: 'Added to your friend.',
    FRIEND_ALREADY_ADDED: 'Already your friend.',
  },
  ko: {
    LOADING: '로딩중...',
    HELLO: '안녕하세요',
    LOGIN: '로그인',
    LOGOUT: '로그아웃',
    EMAIL: '이메일',
    PASSWORD: '패스워드',
    SIGN_UP: '회원가입',
    LOGIN: '로그인',
    FORGOT_PW: '비밀번호를 잊어버리셨나요?',
    FIND_PW: '비밀번호 찾기',
    NAME: '이름',
    STATUS_MSG: '상태메세지',
    REGISTER: '가입',
    SEND_LINK: '링크 보내기',
    FRIEND: '친구',
    MESSAGE: '메세지',
    MY_PROFILE: '나의 프로필',
    UPDATE: '수정',
    NO_CONTENT: '컨텐츠가 없습니다',
    ADD_FRIEND: '친구추가',
    DELETE_FRIEND: '친구삭제',
    SEARCH_USER: '사용자 검색',
    CHAT: '채팅',
    SEND: '보내기',
    WRITE_MESSAGE: '메세지를 입력해주세요…',
    Error: '에러',
    FRIEND_ADDED: '친구로 등록되었습니다.',
    FRIEND_ALREADY_ADDED: '이미 친구입니다.',
  },
});

export const getString = (str) => {
  return strings[str];
};
