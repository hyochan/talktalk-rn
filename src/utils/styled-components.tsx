import * as styledComponents from "styled-components/native";
import ThemeInterface from "./theme";

const {
  default: styled,
  css,
  ThemeProvider
} = styledComponents as styledComponents.ReactNativeThemedStyledComponentsModule<typeof ThemeInterface>;

export { css, ThemeProvider };
export default styled;