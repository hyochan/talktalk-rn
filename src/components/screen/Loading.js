// @flow
import React, { Component } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Image,
  View,
  Animated,
} from 'react-native';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import { Text } from 'react-native-animatable';

import { IC_ICON } from '../../utils/Icons';
import { animateRotateLoop } from '../../utils/Functions';
import { ratio, colors } from '../../utils/Styles';
import { getString } from '../../../STRINGS';

type Styles = {
  container: ViewStyle,
};

const styles: Styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'transparent',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

type Props = {

};
type State = {

};

class Screen extends Component<Props, State> {
  static navigationOptions = {
    title: 'Title',
  };
  spinValue = new Animated.Value(0);
  spin = this.spinValue.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '1260deg'],
  });

  constructor(props: Props) {
    super(props);
  }

  render() {
    return (
      <View style={styles.container}>
        <Animated.Image
          source={IC_ICON}
          style={{
            width: 60,
            height: 60,
            marginBottom: 16,
            transform: [{ rotate: this.spin }],
          }}
        />
        <Text
          animation='fadeIn'
          iterationCount={'infinite'}
          direction='alternate'
          style={{
            color: colors.dodgerBlue,
            fontSize: 16,
          }}
        >
          { getString('LOADING') }
        </Text>
      </View>
    );
  }
}

export default Screen;
