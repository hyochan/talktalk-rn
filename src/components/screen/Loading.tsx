import React, { Component } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Image,
  View,
  Animated,
} from 'react-native';

import { Text } from 'react-native-animatable';

import { IC_ICON } from '../../utils/Icons';
import { animateRotateLoop } from '../../utils/Functions';
import { ratio, colors } from '../../utils/Styles';
import { getString } from '../../../STRINGS';

import styled from 'styled-components/native';

const StyledView = styled.View`
  background-color: transparent;
`;

const StyledContainer = styled.View`
  flex: 1;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

const AnimatedImage = Animated.createAnimatedComponent(Image);
const StyledAnimatedImage = styled(AnimatedImage)`
  width: 60;
  height: 60;
  margin-top: 500;
`;

interface IProps {

}

interface IState {

}

function Screen(props: IProps, state: IState) {
  const spinValue = new Animated.Value(0);

  const spin = spinValue.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '1260deg'],
  });

  return (
    <StyledView>
      <StyledContainer>
        <StyledAnimatedImage
          source={IC_ICON}
          style={{
            transform: [{
              rotate: spin,
            }],
          }}
        />
        <Text
          animation='fadeIn'
          iterationCount={'infinite'}
          direction='alternate'
          style={{
            color: colors.dodgerBlue,
            fontSize: 16,
          }}
        >
          { getString('LOADING') }
        </Text>
      </StyledContainer>
    </StyledView>
  );
}

export default Screen;
