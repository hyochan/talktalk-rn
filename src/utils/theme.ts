import { Dimensions, Platform } from 'react-native';
import { getStatusBarHeight } from 'react-native-status-bar-height';
import { DefaultTheme } from 'styled-components';

const { width, height } = Dimensions.get('window');
let calRatio = width <= height ? 16 * (width / height) : 16 * (height / width);
if (width <= height) {
  if (calRatio < 9) {
    calRatio = width / 9;
  } else {
    calRatio = height / 18;
  }
} else {
  if (calRatio < 9) {
    calRatio = height / 9;
  } else {
    calRatio = width / 18;
  }
}

const theme: DefaultTheme = {
  screenWidth: width,
  screenHeight: height,
  ratio: calRatio / (360 / 9),
  statusBarHeight: getStatusBarHeight(true),
  colors: {
    background: '#e3e3e3',
    dodgerBlue: 'rgb(58,139,255)',
    dusk: 'rgb(65,77,107)',
    blueyGray: 'rgb(134,154,183)',
    cloudyBlue: 'rgb(175,194,219)',
    greenishCyan: 'rgb(80,227,194)',
    paleGray: 'rgb(233,237,244)',
  }
};

export default theme;