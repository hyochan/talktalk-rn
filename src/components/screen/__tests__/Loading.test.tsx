import 'react-native';
import * as React from 'react';
import Loading from '../Loading';
import { ThemeProvider } from 'styled-components/native';
import { getString } from '../../../../STRINGS';
import theme from '../../../utils/theme';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { render, fireEvent, waitForElement } from 'react-native-testing-library';

jest.mock('Animated', () => {
  return {
    loop: jest.fn(() => {
      return {
        start: jest.fn(),
        reset: jest.fn()
      };
    }),
    timing: jest.fn(() => {
      return {
        start: jest.fn(),
      };
    }),
    Value: jest.fn(() => {
      return {
        interpolate: jest.fn(),
      };
    }),
    createAnimatedComponent: (Component) => Component,
  };
});

const props = {
  navigation: {
    navigate: jest.fn(),
  },
  theme
};

const component: React.ReactElement = (
  <ThemeProvider theme={theme}>
    <Loading {...props}/>
  </ThemeProvider>
);

describe('[Loading] rendering test', () => {
  it('renders as expected', async (done) => {
    const { toJSON } = render(component);
    const loading = await waitForElement(() => toJSON());
    expect(loading).toMatchSnapshot();
    done();
  });
});
