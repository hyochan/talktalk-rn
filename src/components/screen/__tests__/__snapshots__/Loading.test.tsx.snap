// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Loading] rendering test renders as expected 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
  testID="loading"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/icons/icon.png",
        }
      }
      style={
        Array [
          Object {
            "height": 60,
            "width": 60,
          },
          Object {
            "transform": Array [
              Object {
                "rotate": undefined,
              },
            ],
          },
        ]
      }
    />
    <Text
      animation="fadeIn"
      direction="alternate"
      iterationCount="infinite"
      style={
        Array [
          Object {
            "color": "rgb(58,139,255)",
            "fontSize": 16,
            "marginTop": 16,
          },
        ]
      }
    >
      LOADING...
    </Text>
  </View>
</View>
`;
